python/paddle/distributed/auto_parallel/api.py

```diff
@@ -1566,3 +1565,182 @@ def unshard_dtensor(dist_tensor):
+
+    def _get_batch(self, batch_data):
+        if isinstance(batch_data, (list, tuple)):
+            if self._all_inputs_in_one_mesh is False:
+                assert len(batch_data) == len(self._meshes)
+            dist_batch_data = []
+            for ind in range(len(batch_data)):
+                shard_dim = (
+                    self._shard_dims[0]
+                    if self._all_inputs_in_one_mesh
+                    else self._shard_dims[ind]
+                )
+                if shard_dim is not None:
+                    placements = [dist.Shard(0)]
+                else:
+                    placements = [dist.Replicate()]
+                mesh = (
+                    self._meshes[0]
+                    if self._all_inputs_in_one_mesh
+                    else self._meshes[ind]
+                )
+                for _ in range(1, len(mesh._shape)):
+                    placements.append(dist.Replicate())
+
+                dist_batch_data.append(
+                    dtensor_from_local(batch_data[ind], mesh, placements)
+                )
+            return dist_batch_data
+        elif isinstance(batch_data, dict):
+            if self._all_inputs_in_one_mesh is False:
+                assert len(self._input_keys) == len(self._meshes)
+            dist_batch_data = {}
+            for ind in len(self._input_keys):
+                key = self._input_keys[ind]
+                shard_dim = (
+                    self._shard_dims[0]
+                    if self._all_inputs_in_one_mesh
+                    else self._shard_dims[ind]
+                )
+                if shard_dim is not None:
+                    placements = [dist.Shard(0)]
+                else:
+                    placements = [dist.Replicate()]
+                mesh = (
+                    self._meshes[0]
+                    if self._all_inputs_in_one_mesh
+                    else self._meshes[ind]
+                )
+                for _ in range(1, len(mesh._shape)):
+                    placements.append(dist.Replicate())
+                dist_batch_data[key] = dtensor_from_local(
+                    batch_data[key], mesh, placements
+                )
+            return dist_batch_data
+        else:
+            raise ValueError(f"Unsupported batch_data type {type(batch_data)}")
+
+    def __next__(self):
+        batch_data = next(self.iter)
+        return self._get_batch(batch_data)
+
+    def __call__(self):
+        return self.__iter__()
+
+
+def shard_dataloader(
+    dataloader: paddle.io.DataLoader,
+    meshes: List[dist.ProcessMesh],
+    input_keys: List[str] = None,
+    shard_dims: Union[list, tuple, str, int] = None,
+    has_dataset_splitted: bool = False,
+) -> ShardDataloader:
+    """
+    Convert the dataloader to a ShardDataloader. See ShardDataloader for details
+
+    Args:
+        dataloader (paddle.io.DataLoader): The dataloader to be sharded.
+        meshes (Union(list, tuple)): The mesh list of the dataloader.
```

Comment:
support type of `PorcessMesh | list[PorcessMesh] | tuple[PorcessMesh]`

Reply:
Done